Add Alibaba Cloud RocketMQ pubsub Doc (#1405)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md

---
type: docs
title: "Alibaba Cloud RocketMQ"
linkTitle: "Alibaba Cloud RocketMQ"
description: "Detailed documentation on the Alibaba Cloud RocketMQ pubsub component"

> aliases:
> - "/operations/components/setup-pubsub/supported-pubsub/setup-alicloud-rocketmq/"

Reviewer: ```suggestion
```

---

## Component format
To setup Alibaba Cloud RocketMQ pubsub create a component of type `pubsub.rocketmq`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration.

> HTTP

Reviewer: ```suggestion
### HTTP
```

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: <NAME>
  namespace: <NAMESPACE>
spec:
  type: pubsub.rocketmq
  version: v1
  metadata:
    - name: accessProto
      value: "http"
    - name: endpoint
      value: "http://********.cn-qingdao-public.aliyuncs.com"
    - name: accessKey
      value: "****************"
    - name: secretKey
      value: "****************"
    - name: consumerGroup
      value: "GID_******"
    - name: instanceId
      value: "MQ_INST_******"
```

> TCP

Reviewer: ```suggestion
### TCP
```

| accessProto                | Y        |sdk proto (HTTP or TCP),default TCP| `"tcp"`
| accessKey                | N        | RocketMQ Credentials| `"****************"`
| secretKey                | N        | RocketMQ Credentials | `"****************"`
| consumerGroup                | N        | consumer group for RocketMQ's subscribers, suggested to provide | `"GID_******"`
| consumerBatchSize                | N        | consumer group for RocketMQ's subscribers, suggested to provide, just for HTTP proto | `1024`
| consumerThreadNums                | N        |consumer group for RocketMQ's subscribers, suggested to provide, just for cgo proto | `20`
| instanceId                | N       | RocketMQ's namespace, optional | `"MQ_INST_******"`
| nameServerDomain                | N        |RocketMQ's name server domain, optional| `"mqrest.cn.aliyuncs.com"`
| retries                | N        | retry times to connect rocketmq's broker, optional | `0`
| content-type                | N        | msg's content-type eg:"application/cloudevents+json; charset=utf-8", application/octet-stream | `"text/plain"`

## Create an AliCloud RocketMQ

> - Follow the instructions [here(en)](https://www.alibabacloud.com/help/doc-detail/200153.htm) on setting up AliCloud RocketMQ
> - Follow the instructions [here(zh-cn)](https://help.aliyun.com/document_detail/200153.html) on setting up AliCloud RocketMQ

Reviewer: ```suggestion
Follow the instructions [here](https://www.alibabacloud.com/help/doc-detail/200153.htm) to setup AliCloud RocketMQ
   - Simplified Chinese translation [here](https://help.aliyun.com/document_detail/200153.html)
```
